spec/runners/jasmine2-test-runner.js

```diff
@@ -0,0 +1,237 @@
+  }
+
+  ['it', 'fit', 'xit'].forEach((key) => {
+    window[key] = (name, originalFn) => {
+      jasmineEnv[key](name, async (done) => {
+        if(originalFn.length === 0) {
+          await originalFn()
+          done();
+        } else {
+          originalFn(done);
+        }
+      });
+    }
+  });
+
+  ['beforeEach', 'afterEach'].forEach((key) => {
+    window[key] = (originalFn) => {
+      jasmineEnv[key](async (done) => {
+        if(originalFn.length === 0) {
+          await originalFn()
+          done();
+        } else {
+          originalFn(done);
+        }
+      })
+    }
+  });
+}
+
+const loadSpecsAndRunThem = (logFile, headless, testPaths) => {
+  return new Promise((resolve) => {
+    const jasmineEnv = jasmine.getEnv();
+
+    // Load before and after hooks, custom matchers
+    require('../helpers/jasmine2-custom-matchers').register(jasmineEnv);
+    require('../helpers/jasmine2-spies').register(jasmineEnv);
+    require('../helpers/jasmine2-time').register(jasmineEnv);
+    require('../helpers/jasmine2-warnings').register(jasmineEnv);
+
+    // Load specs and set spec type
+    for (let testPath of Array.from(testPaths)) { requireSpecs(testPath); }
+    setSpecType('user');
+
+    // Add the reporter and register the promise resolve as a callback
+    jasmineEnv.addReporter(buildReporter({logFile, headless}));
+    jasmineEnv.addReporter(buildRetryReporter(resolve));
+
+    // And finally execute the tests
+    jasmineEnv.execute();
+  })
+}
+
+// This is a helper function to remove a file from the require cache.
+// We are using this to force a re-evaluation of the test files when we need to re-run some flaky tests
+const unrequire = (requiredPath) => {
+  for (const path in require.cache) {
+    if (path === requiredPath) {
+      delete require.cache[path];
+    }
+  }
+}
+
+const requireSpecs = (testPath) => {
+  if (fs.isDirectorySync(testPath)) {
+    for (let testFilePath of fs.listTreeSync(testPath)) {
+      if (/-spec\.js$/.test(testFilePath)) {
+        unrequire(testFilePath);
+        require(testFilePath);
+        // Set spec directory on spec for setting up the project in spec-helper
+        setSpecDirectory(testPath);
+      }
+    }
+  } else {
+    unrequire(testPath);
+    require(testPath);
+    setSpecDirectory(path.dirname(testPath));
+  }
+};
+
+const setSpecField = (name, value) => {
+  const specs = (new jasmine.JsApiReporter({})).specs();
+  if (specs.length === 0) { return; }
+
+  for (let index = specs.length - 1; index >= 0; index--) {
+    if (specs[index][name] != null) { break; }
+    specs[index][name] = value;
+  }
+};
+
+const setSpecType = specType => setSpecField('specType', specType);
+
+const setSpecDirectory = specDirectory => setSpecField('specDirectory', specDirectory);
+
+const buildReporter = ({logFile, headless}) => {
+  if (headless) {
+    return buildConsoleReporter(logFile);
+  } else {
+    const AtomReporter = require('../helpers/atom-reporter.js');
```

Comment:
```suggestion
    const AtomReporter = require('../helpers/jasmine2-atom-reporter.js');
```

I just called my new file `jasmine2-atom-reporter.js` and put it in the same `spec/helpers` directory as `atom-reporter.js`.